jscomp/ml/matching.ml

```diff
@@ -2309,17 +2307,13 @@ let combine_constructor sw_names loc arg ex_pat cstr partial ctx def
         match nonconsts with
           [] -> default
         | _ ->
-            let tag = Ident.create "tag" in
-            let tests =
-              List.fold_right
-                (fun (path, act) rem ->
-                   let ext = transl_extension_path ex_pat.pat_env path in
-                   Lifthenelse(Lprim(extension_slot_eq , [Lvar tag; ext], loc),
-                               act, rem))
-                nonconsts
-                default
-            in
-              Llet(Alias, Pgenval,tag,  arg, tests)
+          List.fold_right
+            (fun (path, act) rem ->
+                let ext = transl_extension_path ex_pat.pat_env path in
+                Lifthenelse(Lprim(extension_slot_eq , [arg; ext], loc),
+                            act, rem))
+            nonconsts
+            default 
```

Comment:
@cristianoc this code is commited by mistake from my investigation on how the try/catch compilation works.

I couldn't figure where the `tag` Llet is used, so I removed it. Could you correct me If I'm wrong.